src/lwc/signals/core.ts

```diff
@@ -70,17 +72,19 @@ function $effect(fn: VoidFunction, options?: Partial<EffectOptions>): Effect {
   const _optionsWithDefaults = { ...defaultEffectOptions, ...options };
   const effectNode: EffectNode = {
     error: null,
-    state: UNSET
+    state: UNSET,
+    stack_depth: 0
   };
 
   const execute = () => {
-    if (effectNode.state === COMPUTING) {
-      throw new Error("Circular dependency detected");
+    if (effectNode.state === COMPUTING && effectNode.stack_depth >= MAX_STACK_DEPTH) {
+      throw new Error(`Circular dependency detected. Maximum stack depth of ${MAX_STACK_DEPTH} exceeded.`);
     }
 
     context.push(execute);
     try {
       effectNode.state = COMPUTING;
+      effectNode.stack_depth++; // Increment depth before execution
```

Comment:
Minor: the code is pretty self explanatory, let's remove the comment as it states the `what` rather than the `why`